hoomd/Communicator.cc

```diff
@@ -1186,6 +1187,32 @@ Communicator::Communicator(std::shared_ptr<SystemDefinition> sysdef,
     m_end.swap(end);
 
     initializeNeighborArrays();
+
```

Comment:
@jglaser Do I also need to implement these changes for the mpcd Communicator?

Reply:
This is recommended, at least if you are communicating arrays of structs, particular non-packed ones, such as combinations of Scalar3's and Scalar4's. The compiler is free to generate holes in the struct and align data types, such as at 16 or 32 byte boundaries. Trying to use gcc's `__attribute__((packed))` results in a lot of warnings. When you try to pass a non-contiguous struct using the `MPI_BYTE` data type, valgrind reports an uninitialized memory error inside the MPI library. This is the perhaps (one of the?) reasons why MPI has structured data types. Fun fact, there are also security concerns associated with communicating uninitialized memory regions over networks.